main.go

```diff
@@ -92,10 +95,131 @@ func startAccepting(maxClients int64, connType, connHost, connPort string, inter
 				LocalPort:  connPort,
 			}
 			clients <- c
+			go reportIPToAbuseIPDB(remoteIpAddr, records, abuseipdbeEnabled)
 		}
 	}()
 }
 
+func reportIPToAbuseIPDB(ip string, records chan<- metrics.RecordEntry, abuseipdbeEnabled bool) {
+	if !abuseipdbeEnabled {
+		return
+	}
+	if isCached(ip) {
+		glog.V(1).Infof("IP is already cached, skipping report")
+		records <- metrics.RecordEntry{
+			RecordType: metrics.RecordEntryTypeReport,
+			IpAddr:     ip,
+			Message:    "IP is already cached, skipping report",
+		}
+		return
+	}
+
+	apiKey := os.Getenv("ABUSE_IPDB_API_KEY")
```

Comment:
Read them in main.go.

I am planning to move the config reading part to a new module. They don't belong here.

Reply:
I have done this, that was an oversight

Reply:
I have done this, that was an oversight

Reply:
I have done this, that was an oversight